api/src/test/java/org/openmrs/api/context/ContextTest.java

```diff
@@ -238,4 +254,111 @@ public void becomeUser_shouldChangeLocaleWhenBecomeAnotherUser() {
 		
 		Context.logout();
 	}
+
+	/**
+	 * @see Context#evictSingleEntity(SessionFactory, Class, String)
+	 */
+	@Test
+	public void evictSingleEntity_shouldClearSingleEntityFromCaches(){
+		//SessionFactory sf = (SessionFactory) applicationContext.getBean("sessionFactory");
+		PersonName name = Context.getPersonService().getPersonName(PERSON_NAME_ID_2);
+		//Load the person so that the names are also stored  in person names collection region
+		Context.getPersonService().getPerson(name.getPerson().getPersonId());
+
+		//Let's have the name in a query cache
+		sf.getCurrentSession().createCriteria(PERSON_NAME_CLASS)
+			.add(Restrictions.eq("personNameId", PERSON_NAME_ID_2))
+			.setCacheable(true)
+			.setCacheRegion(QUERY_REGION)
+			.list();
+		assertTrue(sf.getCache().containsEntity(PERSON_NAME_CLASS, PERSON_NAME_ID_2));
+		assertNotNull(sf.getStatistics().getSecondLevelCacheStatistics(Person.class.getName() + ".names")
```

Comment:
This is failing tests.  I get a NPE which i think is because there are no entities in `"org.openmrs.Person.names"` 
However, i haven't been able to get a likely cause.

cc @wluyima @ibacher

Reply:
what could i be doing wrong here @wluyima 
`sf.getStatistics().getSecondLevelCacheStatistics(Person.class.getName() + ".names").getEntries()` error is returning null 
could you be having suggestions??

Reply:
May be on line 266 the person you are loading with that id doesn't exit or has no names